src/components/dashboard/dashboard.tsx

```diff
@@ -93,263 +92,516 @@ export function Dashboard() {
                 const dateB = parseDate(b.date)
                 return dateB.getTime() - dateA.getTime()
             })
+            setDashboardLinkData(details)
+        } catch (error) {
+            console.log(error)
+        }
+    }
+
+    const fetchRaffleLinkDetails = async (localStorageData: interfaces.ILocalStorageItem[]) => {
+        try {
+            console.log(localStorageData)
+            let details: IDashboardRaffleItemProps[] = []
+            await Promise.all(
+                localStorageData.map(async (item) => {
+                    try {
+                        const _raffleInfo = await peanut.getRaffleInfo({
+                            link: item.link,
+                            baseUrl: `${consts.next_proxy_url}/get-raffle-info`,
+                            APIKey: 'doesnt-matter',
+                        })
+
+                        const x: IDashboardRaffleItemProps = {
+                            hash: item.idx ? item.hash + item.idx : item.hash,
+                            chainId: _raffleInfo.chainId,
+                            totalAmount: '',
+                            token: '',
+                            date: '',
+                            claimed: _raffleInfo.isActive,
+                            link: item.link,
+                            copied: false,
+                            totalSlots: _raffleInfo.totalSlotsNumber,
+                            claimedSlots: _raffleInfo.claimedSlotsNumber,
+                            withMFA: _raffleInfo.withMFA,
+                            withCaptcha: _raffleInfo.withCaptcha,
+                            senderName: _raffleInfo.senderName,
+                            senderAddress: _raffleInfo.senderAddress,
+                        } // TODO: add more details
+
+                        details.push(x)
+                    } catch (error) {
+                        console.log(error)
+                    }
+                })
+            )
 
-            setDashboardData(details)
+            setDashboardRaffleData(details)
         } catch (error) {
             console.log(error)
         }
     }
 
     useEffect(() => {
         if (address) {
-            const data = utils.getAllLinksFromLocalStorage({
+            const linkData = utils.getAllLinksFromLocalStorage({
                 address: address.toString(),
             })
 
-            const filteredData = data?.filter((item) => {
-                return !item.address.includes('saving temp link without depositindex for address')
+            linkData && setLocalStorageLinkData(linkData)
+
+            const raffleData = utils.getAllRaffleLinksFromLocalstorage({
+                address: address.toString(),
             })
 
-            data && setLocalStorageData(filteredData ?? [])
+            raffleData && setLocalStorageRaffleData(raffleData)
+
+            return
         }
         router.prefetch('/send')
     }, [address])
 
     useEffect(() => {
```

Comment:
Shouldn't these also be dependent upoon dashbordLinkData, localStrageRaffleData, dashboardRaffledata?